linkerd/app/src/env.rs

```diff
@@ -730,20 +724,25 @@ fn parse_tap_config(
     match (id_disabled, tap_disabled) {
         (_, true) => Ok(None),
         (true, false) => {
-            error!("{} must be set if identity is disabled", ENV_TAP_DISABLED);
-            Err(EnvError::InvalidEnvVar)
+            let tap_identity = parse(strings, ENV_TAP_SVC_NAME, parse_identity);
+            match tap_identity? {
```

Comment:
Does the tap injector skip pods on which identity is disabled? Would it be better to warn in this case and continue without tap?